drivers/sensor/wsen_hids/wsen_hids.c

```diff
@@ -0,0 +1,272 @@
+	    WE_SUCCESS) {
+		LOG_ERR("Failed to %s sample.", "convert temperature");
+		return -EIO;
+	}
+
+	return 0;
+}
+
+static int hids_channel_get(const struct device *dev, enum sensor_channel channel,
+			    struct sensor_value *value)
+{
+	struct hids_data *data = dev->data;
+	int32_t valueConverted;
+
+	if (channel == SENSOR_CHAN_AMBIENT_TEMP) {
+		valueConverted = (int32_t)data->temperature;
+
+		/* Convert temperature from 0.01 degrees Celsius to degrees Celsius */
+		value->val1 = valueConverted / 100;
+		value->val2 = (valueConverted % 100) * (1000000 / 100);
+	} else if (channel == SENSOR_CHAN_HUMIDITY) {
+		valueConverted = (int32_t)data->humidity;
+
+		/* Convert humidity from 0.01 percent to percent */
+		value->val1 = valueConverted / 100;
+		value->val2 = (valueConverted % 100) * (1000000 / 100);
+	} else {
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+/* Set output data rate. See hids_odr_list for allowed values. */
+static int hids_odr_set(const struct device *dev, const struct sensor_value *odr)
+{
+	struct hids_data *data = dev->data;
+	int odr_index;
+
+	for (odr_index = 0; odr_index < ARRAY_SIZE(hids_odr_list); odr_index++) {
+		if (odr->val1 == hids_odr_list[odr_index].val1 &&
+		    odr->val2 == hids_odr_list[odr_index].val2) {
+			break;
+		}
+	}
+
+	if (odr_index == ARRAY_SIZE(hids_odr_list)) {
+		/* ODR not allowed (was not found in hids_odr_list) */
+		LOG_ERR("Bad sampling frequency %d.%d", odr->val1, odr->val2);
+		return -EINVAL;
+	}
+
+	if (WE_SUCCESS !=
+	    HIDS_setOutputDataRate(&data->sensor_interface, (HIDS_outputDataRate_t)odr_index)) {
+		LOG_ERR("Failed to set output data rate");
+		return -EIO;
+	}
+
+	return 0;
+}
+
+static int hids_attr_set(const struct device *dev, enum sensor_channel chan,
+			 enum sensor_attribute attr, const struct sensor_value *val)
+{
+	if (chan != SENSOR_CHAN_ALL) {
+		LOG_WRN("attr_set() is not supported on channel %d.", chan);
+		return -ENOTSUP;
+	}
+
+	if (attr == SENSOR_ATTR_SAMPLING_FREQUENCY) {
+		return hids_odr_set(dev, val);
+	} else {
+		return -ENOTSUP;
+	}
+}
+
+static const struct sensor_driver_api hids_driver_api = {
+	.attr_set = hids_attr_set,
+#if CONFIG_WSEN_HIDS_TRIGGER
+	.trigger_set = hids_trigger_set,
+#endif
+	.sample_fetch = hids_sample_fetch,
+	.channel_get = hids_channel_get,
+};
+
+int hids_init(const struct device *dev)
+{
+	const struct hids_config *config = dev->config;
+	struct hids_data *data = dev->data;
+	uint8_t device_id;
+
+	/* Initialize WE sensor interface */
+	WE_sensorInterfaceType_t interface_type = data->sensor_interface.interfaceType;
+
+	HIDS_getDefaultInterface(&data->sensor_interface);
+	data->sensor_interface.interfaceType = interface_type;
+	if (data->sensor_interface.interfaceType == WE_i2c) {
+#if DT_ANY_INST_ON_BUS_STATUS_OKAY(i2c)
+		data->sensor_interface.handle = (void *)&config->bus_cfg.i2c;
+#endif
+	} else {
+#if DT_ANY_INST_ON_BUS_STATUS_OKAY(spi)
+		data->sensor_interface.handle = (void *)&config->bus_cfg.spi;
+#endif
```

Comment:
This conditional logic isn't quite right - you want to test if *this* instance is on an I2C or SPI bus, not *any* instance.

Reply:
This still needs to be addressed

Reply:
@MaureenHelm Thanks for the feedback. However, I think in this case the logic is in fact correct, as there is a surrounding if-statement that checks if the instance is on an I2C or SPI bus. `DT_ANY_INST_ON_BUS_STATUS_OKAY` is used, as the corresponding field of the `config->bus_cfg` union is included in the compilation based on the same logic.

Reply:
Yes, that does make sense. Changed it.